### PR TITLE
improve global handling

### PR DESCRIPTION
**Pull request by calvinmetcalf**
does 2 things

1. fixes the global shim so it works in a web worker
2. sets a flag so it can be turned off (really useful for me to be able to do this).

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -146,7 +149,7 @@ export default function commonjs ( options = {} ) {
 						return;
 					}
 
-					if ( node.type === 'ThisExpression' && scopeDepth === 0 ) {
+					if ( node.type === 'ThisExpression' && scopeDepth === 0 && !ignoreGlobal) {
```
Comment (TrySound) [inline src/index.js]: Space after `ignoreGlobal`

Comment (calvinmetcalf) [inline src/index.js]: fixed

File: test/test.js [hunk 2]
```diff
@@ -222,4 +222,23 @@ describe( 'rollup-plugin-commonjs', () => {
 			plugins: [ commonjs() ]
 		}).then( executeBundle );
 	});
+	it( 'can ignore references to `global`', () => {
```
Comment (TrySound) [inline test/test.js]: new line before it

File: src/index.js [hunk 3]
```diff
@@ -230,7 +233,7 @@ export default function commonjs ( options = {} ) {
 			var intros = [];
 
 			if ( bundleUsesGlobal ) {
-				intros.push( `var __commonjs_global = typeof window !== 'undefined' ? window : typeof global !== 'undefined' ? global : this;` );
+				intros.push( `var __commonjs_global = typeof window !== 'undefined' ? window : typeof global !== 'undefined' ? global :  typeof self !== 'undefined' ? self : this;` );
```
Comment (TrySound) [inline src/index.js]: What is `self`?

Comment (TrySound) [inline src/index.js]: I see, it's or web workers. Can you move it in separate PR also with some tests (is it possible somehow)?

**Comment by TrySound** [comment]
And squash commits please. You can rename commit with issues.

**Comment by calvinmetcalf** [comment]
yeah I can separate the things out, squash and rename tommorow

**Comment by calvinmetcalf** [comment]
rebased and sqashed and the web worker stuff removed

**Comment by TrySound** [comment]
@calvinmetcalf Great. Thanks. I think in future we should remove global detecting in favour of your node-globals
